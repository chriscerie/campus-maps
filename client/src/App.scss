.App {
  text-align: center;
  font-family: 'Poppins', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 60%;
}

.App h2 {
  font-size: 1.6em;
  color: #2537ff;
  margin-top: 0;
  margin-bottom: 0.3em;
}

.App p {
  font-size: 0.95em;
  margin-top: 0;
  margin-bottom: 0.3em;
}

.App ul {
  padding: 0;
  margin: 0;
  list-style: none;
}

.App li {
  list-style: none;
}

.App label {
  display: inline-block;
  font-weight: bold;
  margin-bottom: 0.5em;
}

@media (min-width: 481px) {
  .App {
    font-size: 70%;
  }
}

@media (min-width: 769px) {
  .App {
    font-size: 80%;
  }
}

@media (min-width: 1025px) {
  .App {
    font-size: 90%;
  }
}

@media (min-width: 1201px) {
  .App {
    font-size: 100%;
  }
}
