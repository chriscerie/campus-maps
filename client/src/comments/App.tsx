import Comments from './components/Comments';

const App = () => {
    return (
        <div>
            <Comments/>
        </div>
    );
};

export default App;