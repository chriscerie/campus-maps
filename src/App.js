import React, { useState } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import CheckingSignedIn from "./pages/CheckingSignedIn";
import Home from "./pages/Home";
import Profile from "./pages/Profile";
import Private from "./pages/Private";
import PageNotFound from "./pages/PageNotFound";

export default function App() {
  const [isSignedIn, setIsSignedIn] = useState(null);
  const script = document.createElement("script");
  script.src = "https://apis.google.com/js/platform.js";
  script.onload = () => initGoogleSignIn();
  document.body.appendChild(script);

  function initGoogleSignIn() {
    window.gapi.load("auth2", () => {
      window.gapi.auth2
        .init({
          client_id: process.env.REACT_APP_AUTH_CLIENT_ID,
        })
        .then(() => {
          const authInstance = window.gapi.auth2.getAuthInstance();
          const isSignedIn = authInstance.isSignedIn.get();
          setIsSignedIn(isSignedIn);

          authInstance.isSignedIn.listen((isSignedIn) => {
            setIsSignedIn(isSignedIn);
          });
        });
    });
    window.gapi.load("signin2", () => {
      window.gapi.signin2.render("login-button", {
        theme: "dark",
      });
    });
  }

  function PrivateRoute(props) {
    const { component, ...rest } = props;
    return <Route {...rest} component={isSignedIn ? component : Private} />;
  }

  if (isSignedIn !== null) {
    return (
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={Home} />
          <PrivateRoute exact path="/profile" component={Profile} />
          <Route path="/" component={PageNotFound} />
        </Switch>
      </BrowserRouter>
    );
  }

  return <CheckingSignedIn />;
}
