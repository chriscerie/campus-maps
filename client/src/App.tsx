import { useEffect, useState } from 'react';
import './App.scss';
import HomePage from './pages/HomePage';

// Fetching the GET route from the Express server which matches the GET route from server.js
const callBackendAPI = async () => {
  const response = await fetch('/express_backend');
  const body = await response.json();

  if (response.status !== 200) {
    throw Error(body.message);
  }
  return body;
};

function App() {
  const [data, setData] = useState<{ data: string }>({
    data: 'Waiting for data.',
  });

  useEffect(() => {
    callBackendAPI()
      .then((res) => setData({ data: res.message }))
      .catch((err) => console.log(err));
  }, []);

  return (
    <div className="App">
      <HomePage />
    </div>
  );
}

export default App;
