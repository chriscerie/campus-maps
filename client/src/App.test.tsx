import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from './reducers';
import App from './App';

describe('App', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
    renderer.unmount();
  });
});
