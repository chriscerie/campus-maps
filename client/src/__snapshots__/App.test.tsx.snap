// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render successfully 1`] = `
<div
  className="App"
>
  <HomePage />
</div>
`;
